surefire-api/src/main/java/org/apache/maven/surefire/testset/TestListResolver.java

```diff
@@ -479,6 +481,27 @@ else if ( hasMethod )
             updatedFilters( isExcluded, test, patterns, includedFilters, excludedFilters );
         }
     }
+    
+    private static void regexSanityCheck( String request, String classNameExpr, String methodNameExpr )
+    {
+        // this will emit any exception if the regex is not valid at all
+        Pattern.compile( request );
+        try
+        {
+            Pattern.compile( classNameExpr );
+            Pattern.compile( methodNameExpr );
+            if ( methodNameExpr.indexOf( "#" ) != -1 )
```

Comment:
I thought you would use `String.contains` in both constructors of `ResolvedTest`.
I think we do not need to use `Pattern.compile` here because this is already done by `maven-shared-utils` anyway.
Lets use the sanity check if the `isRegex` is set, because I do not know how Cucumber would use JUnit Description of method or class name without regex. Cucumber is different runner from most of the other.